tests/unittests/test_tabular.py

```diff
@@ -28,75 +28,93 @@
-                  'problem_type': REGRESSION,
-                  'label_column': 'SalePrice',
-                  'performance_val': 0.076} # Regression with mixed feature-types, skewed Y-values.
-
-toyregres_dataset = {'url': 'https://autogluon.s3-us-west-2.amazonaws.com/datasets/toyRegression.zip', 
-                     'name': 'toyRegression',
-                     'problem_type': REGRESSION, 
-                    'label_column': 'y', 
-                    'performance_val': 0.183}
-# 1-D toy deterministic regression task with: heavy label+feature missingness, extra distraction column in test data
-
-toyclassif_dataset = {'url': 'https://autogluon.s3-us-west-2.amazonaws.com/datasets/toyClassification.zip',
-                     'name': 'toyClassification',
-                     'problem_type': MULTICLASS, 
-                    'label_column': 'y', 
-                    'performance_val': 0.436}
-# 2-D toy noisy, imbalanced 4-class classification task with: feature missingness, out-of-vocabulary feature categories in test data, out-of-vocabulary labels in test data, training column missing from test data, extra distraction columns in test data
-# toyclassif_dataset should produce 3 warnings:
-# UserWarning: These columns from this dataset were not present in the training dataset (AutoGluon will ignore them):  ['distractioncolumn1', 'distractioncolumn2']
-# UserWarning: The columns listed below from the training data are no longer in the given dataset. (AutoGluon will proceed assuming their values are missing, but you should remove these columns from training dataset and train a new model):  ['lostcolumn']
-# UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
-
-
- # List containing dicts for each dataset to include in benchmark (try to order based on runtimes)
-datasets = [toyregres_dataset, toyclassif_dataset, binary_dataset, regression_dataset, multi_dataset]
 
 def test_tabular():
+    ############ Benchmark options you can set: ########################
+    perf_threshold = 1.1 # How much worse can performance on each dataset be vs previous performance without warning
+    seed_val = 0 # random seed
+    subsample_size = None
+    hyperparameter_tune = False
+    verbosity = 2 # how much output to print
+    hyperparameters = None
+    time_limits = None
+    fast_benchmark = True # False
+    # If True, run a faster benchmark (subsample training sets, less epochs, etc),
+    # otherwise we run full benchmark with default AutoGluon settings.
+    # performance_value warnings are disabled when fast_benchmark = True.
+
+    #### If fast_benchmark = True, can control model training time here. Only used if fast_benchmark=True ####
+    if fast_benchmark:
+        subsample_size = 100
+        nn_options = {'num_epochs': 3} 
+        gbm_options = {'num_boost_round': 30}
+        hyperparameters = {'GBM': gbm_options, 'NN': nn_options}
```

Comment:
Maybe we should start using the default hyperparameters to ensure all default models are working properly? This test wouldn't catch a crash in KNN for example.

Reply:
will add one unit-test that runs with default hyperparams. The problem is default may be slow, depending on models/dataset